File: cs50x-2022/week6/sentimental-cash/cash.py
from cs50 import get_float

# Get the user's input
while True:
    dollars = get_float("Change owed: ")
    if (dollars > 0):
        break

# Convert dollars to cents
cents = round(100 * dollars)

# Number of coins
coins = 0

while cents != 0:
    if cents >= 25:
        cents -= 25
    elif cents >= 10:
        cents -= 10
    elif cents >= 5:
        cents -= 5
    else:
        cents -= 1
    coins += 1

print(coins)

File: cs50x-2022/week1/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{   
    // Get the user's input
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    
    // Convert dollars to cents
    int cents = round(100 * dollars);
    
    // Number of coins
    unsigned short coins = 0;
    
    while (cents != 0)
    {
        if (cents >= 25)
        {
            cents -= 25;   
        }
        else if (cents >= 10)
        {
            cents -= 10;         
        }
        else if (cents >= 5)
        {
            cents -= 5;      
        }
        else
        {
            cents--;      
        }
        coins++;
    }
    printf("%i\n", coins);
}